app/src/main/java/com/example/bookwormadventuresdeluxe2/MyProfileFragment.java

```diff
@@ -116,18 +185,30 @@ public void onClick(View view)
                         intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK
                                 | Intent.FLAG_ACTIVITY_CLEAR_TOP
                                 | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+
                         /* Take User back to Login Page */
                         startActivity(intent);
                     }
+
                 default:
                     /* Unexpected resource id*/
                     throw new Exception("Unexpected resource Id inside click listener."
-                            + "Expected: R.id.login_button Or R.idcreate_account_button");
+                            + "Expected: R.id.login_button Or R.id create_account_button");
             }
-        } catch (Exception e)
+        }
+        catch (Exception e)
         {
             /* Log message to debug*/
             Log.d(TAG, e.getMessage());
         }
     }
+
+    /**
+     * Callback for UserProfileObject
+     */
+    @Override
```

Comment:
MyProfileFragment does not have to implement FirebaseUserGetSet.UserCallback right?
Since nothing is done here.

Reply:
noo that is the interface way. for communicating between two classes at the same time

Reply:
it will return a null object without OnCallBack because the code runs past the OnSuccessListener before query is finished

Reply:
Interesting. It was successfully editing when I removed this empty override and the implements FirebaseUserGetSet.UserCallback from the top.

Reply:
hmm. I am 100% sure it needs it for OnSuccess returns. But you may be right. There are plenty examples of callback like this v.
![image](https://user-images.githubusercontent.com/39696855/97255260-d7bea300-17d5-11eb-8a7d-12bc9c4da548.png)

Reply:
ok. thanks for sharing